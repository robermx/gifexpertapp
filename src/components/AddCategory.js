import React, { useState } from 'react';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');
  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handelSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length >= 2) {
      setCategories((cat) => [inputValue, ...cat]);
      setInputValue('');
    }
  };
  return (
    <form onSubmit={handelSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};
