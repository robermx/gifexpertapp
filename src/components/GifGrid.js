// import React, { useState, useEffect } from 'react';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="text-light animate__animated animate__fadeInDown">
        {category}
      </h3>
      {loading && (
        <p className="center animate__animated animate__pulse">Cargando</p>
      )}

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
