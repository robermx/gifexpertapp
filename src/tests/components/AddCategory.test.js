import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas para AddCategory component', () => {
  const setCategories = jest.fn();
  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('Mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('cambios en la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'bachas';
    input.simulate('change', { target: { value } });
  });

  test('No debe de postear la informaciÃ³n con submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('LLamar SetCategories y limpiar a caja de texto', () => {
    const value = 'bachas';
    wrapper.find('input').simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
