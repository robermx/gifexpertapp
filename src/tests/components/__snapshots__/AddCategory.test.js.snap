// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para AddCategory component Mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
